composer.json

```diff
@@ -25,7 +23,10 @@
 		"staudenmeir/laravel-upsert": "^1.3",
 		"cweagans/composer-patches": "^1.7",
 		"composer/semver": "*",
-		"php81_bc/strftime": "^0.5.0"
+		"php81_bc/strftime": "^0.5.0",
+		"components/jquery": "3.7.1",
+		"jquery/ui": "1.13.3",
+		"jquery/validation": "1.19.5"
```

Comment:
That's the implementation we used for the main branch. I initially made similar changes with versions 3.3 and 3.4, but we encountered some potential issues.

Typically, the installation process should be to run `composer install` followed by `npm run build`. However, if a user runs `npm run build` first (which copies files from `node_modules` to existing paths), and then runs `composer install` (which removes these dependencies since they should now be sourced from `npm`), it can cause problems. The copied directories from `npm run build` get deleted by Composer. To prevent this and streamline the process into a single installation step, we decided to manage it through a Composer script. This ensures that everything is properly installed and copied to the correct paths. And it also ensures that we don't introduce breaking changes for 3.3 and 3.4. 

However, the implementation for main branch is what you suggested. These js dependencies are handled via npm and copying the files are handled during build. Here's a link of the PR for main if you'd like to check it: https://github.com/pkp/pkp-lib/pull/10168

Reply:
Thanks for the information, I'll mark it as resolved ☺